src/OSPSuite.Presentation/Presenters/ParameterIdentifications/ParameterIdentificationFeedbackPresentersManager.cs

```diff
@@ -0,0 +1,59 @@
+﻿using OSPSuite.Core.Domain.ParameterIdentifications;
+using OSPSuite.Core.Events;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Container;
+using OSPSuite.Utility.Events;
+
+namespace OSPSuite.Presentation.Presenters.ParameterIdentifications
+{
+   public interface IParameterIdentificationFeedbackPresentersManager : 
+      IListener<ParameterIdentificationStartedEvent>,
+      IListener<ParameterIdentificationTerminatedEvent>,
+      IListener<ProjectClosedEvent>
+   {
+      IParameterIdentificationFeedbackPresenter FeedbackPresenterFor(ParameterIdentification parameterIdentification);
+   }
+
+   public class ParameterIdentificationFeedbackPresentersManager : IParameterIdentificationFeedbackPresentersManager
+   {
+      private readonly IContainer _container;
+      private readonly Cache<ParameterIdentification, IParameterIdentificationFeedbackPresenter> _currentParameterIdentifications = new Cache<ParameterIdentification, IParameterIdentificationFeedbackPresenter>();
+      private readonly object _locker = new object();
+
+      public ParameterIdentificationFeedbackPresentersManager(IContainer container)
+      {
+         _container = container;
+      }
+
+      public IParameterIdentificationFeedbackPresenter FeedbackPresenterFor(ParameterIdentification parameterIdentification)
+      {
+         lock (_locker)
+         {
+            if (_currentParameterIdentifications.Contains(parameterIdentification))
+               return _currentParameterIdentifications[parameterIdentification];
+
+            var presenter = _container.Resolve<IParameterIdentificationFeedbackPresenter>();
+            _currentParameterIdentifications.Add(parameterIdentification, presenter);
+            return presenter;
+         }
+      }
+
+      public void Handle(ParameterIdentificationStartedEvent eventToHandle)
+      {
+         FeedbackPresenterFor(eventToHandle.ParameterIdentification).ParameterIdentificationStarted(eventToHandle.ParameterIdentification);
+      }
+
+      public void Handle(ParameterIdentificationTerminatedEvent eventToHandle)
+      {
+         FeedbackPresenterFor(eventToHandle.ParameterIdentification).ParameterIdentificationTerminated(eventToHandle.ParameterIdentification);
+      }
+
+      public void Handle(ProjectClosedEvent eventToHandle)
+      {
+         lock (_locker)
+         {
+            _currentParameterIdentifications.Clear();
```

Comment:
You are holding a reference all the `IParameterIdentificationFeedbackPresenter`s here.
They need to be `ReleaseFrom` explicitly as they are registered in the EventPublisher on construction. This is a memory leak 

so something like 
`_currentParameterIdentifications.Each(x=>x.ReleaseFrom(_eventPublisher));` where _eventPublisher is injected into the constructor of the Manager

Reply:
I am wondering if ApplicationController is not doing the work already?

```
      public virtual ISingleStartPresenter Open<TSubject>(TSubject subject, ICommandCollector commandCollector)
      {
         var presenterToOpen = PresenterFor(subject);
         if (presenterToOpen == null)
         {
            presenterToOpen = CreatePresenterForSubject(subject);
            presenterToOpen.Closing += removePresenter;
            presenterToOpen.InitializeWith(commandCollector);
            _allOpenedPresenters.Add(presenterToOpen);
         }
         return presenterToOpen;
      }
```

Reply:
You are not resolving the presenter usting the applicationController but you are using the container. 

But this is in fact a good point. I am wondering if we need this ParameterIdentifiactionManager at all since we already have everything in the applicationController. Why aren't you using this?

Reply:
Yeah the ApplictionController does exactly what you need I feel for almost everything. Let's talk about it during the call later today